ecommerce/api.py

```diff
@@ -244,3 +270,43 @@ def make_reference_id(order):
             A reference number for use with CyberSource to keep track of orders
     """
     return "{}{}-{}".format(_REFERENCE_NUMBER_PREFIX, settings.CYBERSOURCE_REFERENCE_PREFIX, order.id)
+
+
+def complete_order(order):
+    """
+    Once an order is fulfilled, we need to create an enrollment and notify other services.
+
+    Args:
+        order (Order): An order which has just been fulfilled
+    """
+    run = order.get_bootcamp_run()
+    if is_paid_in_full(user=order.user, bootcamp_run=run):
+        BootcampRunEnrollment.objects.get_or_create(
+            user=order.user,
+            bootcamp_run=run,
+        )
+
+        try:
+            application = order.applications.get()
+            try:
+                application.complete()
+                application.save()
+            except TransitionNotAllowed:
+                log.exception(
+                    "Application received full payment but state cannot transition to COMPLETE from %s for order %d",
+                    application.state,
+                    order.id,
+                )
+        except BootcampApplication.DoesNotExist:
+            log.exception("Missing application for order %d. Unable to set application state.", order.id)
+
+    try:
```

Comment:
If it's easy enough to just remove this try/accept, I think we should do it. Matt has an open PR to remove this anyway

Reply:
I'm happy to have it removed but I would rather wait until @mbertrand's PR because it is a major change in behavior

Reply:
Yup, sounds good